project/Build.scala

```diff
@@ -1512,7 +1512,19 @@ object Build {
         throw new MessageOnlyException(
             "testingExample/test is not supported because it requires DOM " +
             "support. Use testingExample/testHtml instead.")
-      }
+      },
+
+      // HACK: Test Module support.
+      scalaJSLinkerConfig ~= { _
+        .withModuleKind(ModuleKind.ESModule)
+        .withModuleSplitStyle(ModuleSplitStyle.SmallestModules)
+      },
+      jsEnv in Test := {
+        val out = (artifactPath in Test in fastOptJS).value
+        val config = NodeJSEnv.Config()
+          .withEnv(Map("NODE_PATH" -> out.toString))
```

Comment:
That seems fishy. Should we emit references to other modules as `./foo.js` instead of `foo.js` to avoid the need for `NODE_PATH`?

Reply:
Maybe. I didn't understand that standard well enough. Is that required to work in every engine? (if yes, we should probably indeed do that).

Reply:
Nothing is required to work in every engine, if you only read the standard. But if an engine doesn't support explicit relative paths like that, it's unlikely that it supports any kind of reasonable module resolution anyway.

Reply:
Hmmm... But then haven't we taken the stance that we do not want to rely on specifics of engines? Maybe we need to have an additional `moduleName` function (just like `jsNameWithSourceMap`, or have that one return all of the things we need).

Reply:
I think I like this idea much more. We should maybe even do this for the references between source maps and JS file. That way:
- We can provide full backwards compatibility.
- We have a single, simple mechanism to deal with all the oddities of how things reference each other in a given environment.